src/storage/store/dictionary_column.cpp

```diff
@@ -28,19 +29,21 @@ DictionaryColumn::DictionaryColumn(const std::string& name, const MetadataDAHInf
 void DictionaryColumn::initChunkState(Transaction* transaction, node_group_idx_t nodeGroupIdx,
     Column::ChunkState& readState) {
     // We put states for data and offset columns into childrenStates.
-    readState.childrenStates.resize(2);
+    if (readState.childrenStates.size() < 2) {
```

Comment:
I don't think this `if` check is doing much here. What you really want is to KU_ASSERT that the size of children states are matching the COUNT, as the child states should already be resized.

Reply:
updated